btrack/models.py

```diff
@@ -18,20 +18,25 @@
 __email__ = "code@arlowe.co.uk"
 
 
-import dataclasses
-import os
 from typing import List, Optional
 
 import numpy as np
+from pydantic import BaseModel, root_validator, validator
 
-from . import constants, utils
+from . import constants
 from .optimise.hypothesis import H_TYPES, PyHypothesisParams
 
 __all__ = ["MotionModel", "ObjectModel", "HypothesisModel"]
 
 
-@dataclasses.dataclass
-class MotionModel:
+def _check_symmetric(
+    x: np.array, rtol: float = 1e-5, atol: float = 1e-8
+) -> bool:
+    """Check that a matrix is symmetric by comparing with it's own transpose."""
+    return np.allclose(x, x.T, rtol=rtol, atol=atol)
+
+
+class MotionModel(BaseModel):
```

Comment:
So we can leverage all the nice type checking and validators from pydantic - although, I guess it could also be a pydantic dataclass?

Reply:
Looks like that should be doable https://pydantic-docs.helpmanual.io/usage/dataclasses/

Reply:
I'm inclined to leave these "as is" for the moment, since it would require quite a bit of reworking by the look of it.  The `BaseModel` has a decent `__repr__` anyway.  Perhaps this could be a follow-up PR?

Reply:
Sure